src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunGenericHelperNode.cs

```diff
@@ -73,6 +73,39 @@ protected sealed override void OnMarked(NodeFactory factory)
             }
         }
 
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory)
+        {
+            DependencyList dependencies = new DependencyNodeCore<NodeFactory>.DependencyList();
+
+            // When methods use generic lookup nodes, make sure they're also available as template dictionary nodes
+            if (_dictionaryOwner is MethodDesc)
+            {
+                dependencies.Add(factory.NativeLayout.TemplateMethodLayout((MethodDesc)_dictionaryOwner), "Type loader template");
+                dependencies.Add(_lookupSignature.TemplateDictionaryNode(factory), "Type loader template");
```

Comment:
We really shouldn't depend on the TemplateMethodLayout/TemplateTypeLayout directly as a result of having code which is shared generic. We should depend on those when we actually have some specific desire for the template to exist. For now though, this is ok, as we enable reflection for all code we emit (Later on we should track this based on what we've decided to make reflectable. (I'm ok with those being added here for now, but it should be under some sort of comment that indicates why that isn't a great idea long term.)

For the _lookupSignature.TemplateDictionaryNode dependency, we should track that dependency as a ConditionalStaticDependency which is only added if the TemplateMethodLayout/TemplateTypeLayout are in the graph.